Source/Tools/FEXRootFSFetcher/Main.cpp

```diff
@@ -555,8 +555,8 @@ std::optional<std::vector<FileTargets>> GetRootFSLinks() {
   const json_t* json = FEX::JSON::CreateJSON(Data, Pool);
 
   if (!json) {
-    fprintf(stderr, "Couldn't create json");
-    return {};
+    fmt::print(stderr, "Failed to parse JSON from RootFSLinks file '{}' - invalid JSON format", Data);
+    __builtin_unreachable();
```

Comment:
The unreachable annotation makes these branches undefined behavior as an optimization hint to the compiler (ironically allowing the print to be removed entirely), which isn't what we're looking for here.

Sadly FEX's error handling isn't particularly versatile, but either `std::abort` or `ERROR_AND_DIE_FMT` are the closest in behavior here.

Reply:
Good point. Hummm, then I guess I shouldn't be using FEX_UNREACHABLE either.

Reply:
Ah, no... the others are fine because EFmt is really meant to abort after printing the message so the unreachable is fine.

Now fixed.

Reply:
> Ah, no... the others are fine because EFmt is really meant to abort after printing the message so the unreachable is fine.

This is really misleading and fragile though. Why not just clearly abort in these cases too, where there's no need to bring UB into play?

Reply:
`AFmt` does

Reply:
But also only in builds with `ASSERTIONS_ENABLED` enabled. Needs `ERROR_AND_DIE_FMT` to actually assert in release builds.

Reply:
yeah - i misread this:
```
#define ERROR_AND_DIE_FMT(...)      \
  do {                              \
    LogMan::Msg::EFmt(__VA_ARGS__); \
    FEX_TRAP_EXECUTION;             \
  } while (0)
```

I skimmed this code and assumed EFmt would call FEX_TRAP_EXECUTION (which in turn calls ForcedAssert). Thanks for pointing this out.